Query ip

dannybackx opened the pull request:
Minimal testing performed :-)

This goes with a PR for el-client that I need to create too.

Functionality provided with this PR is best summarized by the additional codes in cmd/cmd.h : 
+  CMD_GET_WIFI_INFO,   // query ip address info
+  CMD_SET_WIFI_INFO,   // set ip address info
+  CMD_WIFI_GET_APCOUNT = 50,    // Get number of access points
+  CMD_WIFI_GET_APNAME,     // Get the name (ssid) of an access point
+  CMD_WIFI_SELECT_SSID,     // select a network to connect to
+  CMD_WIFI_SIGNAL_STRENGTH,    // Query RSSI (wifi signal strength)
+  CMD_WIFI_GET_SSID,           // Query SSID/network name currently connected to
+  CMD_WIFI_START_SCAN,         // Trigger a scan (takes a long time)

Commit:
Following tve's guidelines to submit the changes for query-ip, I think.

Commit:
More includes

dannybackx committed:
Need these to compile

dannybackx committed:
Looks like I missed a -1 to 255 "conversion".

tve commented on cmd/handlers.c:
```diff
@@ -197,3 +245,114 @@ cmdAddCallback(CmdPacket *cmd) {
+// Query the number of wifi access points
+static void ICACHE_FLASH_ATTR cmdWifiGetApCount(CmdPacket *cmd) {
+  int n = wifiGetApCount();
+  os_printf("WifiGetApCount : %d\n", n);
```
The `os_printf` in this file need to be converted to `DBG` so they can be disabled. See 8 lines up, for example.

tve commented on cmd/handlers.c:
```diff
@@ -197,3 +245,114 @@ cmdAddCallback(CmdPacket *cmd) {
+    uint8_t ix;
+    cmdPopArg(&req, &ix, 1);
+    len = cmdArgLen(&req);
+    pass = (char *)os_malloc(len+2);
```
Shouldn't that be len+1 ?

tve commented on cmd/handlers.c:
```diff
@@ -197,3 +245,114 @@ cmdAddCallback(CmdPacket *cmd) {
+ * Once we're attached to some wireless network, choose not to pick up address from
+ * DHCP or so but set our own.
+ */
+static void ICACHE_FLASH_ATTR cmdSetWifiInfo(CmdPacket *cmd) {
```
unimplemented? then remove for now?

tve commented on esp-link/cgiservices.c:
```diff
@@ -1,4 +1,5 @@
 #include <esp8266.h>
+#include "cmd.h"
```
why is this needed?

tve commented on esp-link/cgiwifi.c:
```diff
@@ -238,6 +241,16 @@ static int ICACHE_FLASH_ATTR cgiWiFiStartScan(HttpdConnData *connData) {
   return HTTPD_CGI_DONE;
 }
 
+// Start scanning, API interface
```
Can you consolidate the code a bit and have one function call the other? Flash isn't infinite...

tve commented on esp-link/cgiwifi.c:
```diff
@@ -945,3 +964,58 @@ void ICACHE_FLASH_ATTR wifiInit() {
+  if (ptr != 0)
+    strncpy(ptr, cgiWifiAps.apData[i]->ssid, 32);
+
+  os_printf("AP %s\n", cgiWifiAps.apData[i]->ssid);
```
DBG or delete...

tve commented on esp-link/cgiwifi.c:
```diff
@@ -945,3 +964,58 @@ void ICACHE_FLASH_ATTR wifiInit() {
+// But it's good to have similar functionality close to each other.
+// This performs functions similar to cgiWiFiConnect()
+int ICACHE_FLASH_ATTR wifiConnect(char *ssid, char *pass) {
+// Danny
```
unimplemented -> remove for now?

tve commented on esp-link/cgiwifi.c:
```diff
@@ -945,3 +964,58 @@ void ICACHE_FLASH_ATTR wifiInit() {
+  return cgiWifiAps.noAps;
+}
+
+ICACHE_FLASH_ATTR void wifiGetApName(int i, char *ptr) {
```
Add a comment for the function?

tve commented on esp-link/cgiwifi.c:
```diff
@@ -945,3 +964,58 @@ void ICACHE_FLASH_ATTR wifiInit() {
+  return 0;
+}
+
+ICACHE_FLASH_ATTR int wifiSignalStrength(int i) { 
```
Add a comment for the function?

tve commented on esp-link/cgiwifi.c:
```diff
@@ -945,3 +964,58 @@ void ICACHE_FLASH_ATTR wifiInit() {
+  return rssi;
+}
+
+void ICACHE_FLASH_ATTR cmdWifiQuerySSID(CmdPacket *cmd) {
```
This should be in handlers.c not here...

tve commented on esp-link/cgiwifi.c:
```diff
@@ -945,3 +964,58 @@ void ICACHE_FLASH_ATTR wifiInit() {
+  uint32_t callback = req.cmd->value;
+
+  struct station_config conf;
+  // bool res = wifi_station_get_config(&conf);
```
???

tve commented on cmd/handlers.c:
```diff
@@ -197,3 +245,114 @@ cmdAddCallback(CmdPacket *cmd) {
+  os_printf("SetWifiInfo()\n");
+}
+
+static void ICACHE_FLASH_ATTR cmdWifiSignalStrength(CmdPacket *cmd) {
```
Add a comment for the function?

tve commented on esp-link/status.c:
```diff
@@ -3,6 +3,7 @@
 #include <esp8266.h>
 #include "config.h"
 #include "serled.h"
+#include "cmd.h"
```
Why is this needed?

tve commented on esp-link/mqtt_client.c:
```diff
@@ -114,4 +116,23 @@ mqtt_client_on_data(MqttDataCallback dataCb) {
 }
 
+// Command handler for MQTT information
+void ICACHE_FLASH_ATTR cmdMqttGetClientId(CmdPacket *cmd) {
```
Should be in handlers.c?

tve commented on esp-link/mqtt_client.c:
```diff
@@ -1,9 +1,11 @@
 #ifdef MQTT
 #include <esp8266.h>
+#include "cmd/cmd.h"
```
This should be included here.

tve commented on esp-link/main.c:
```diff
@@ -13,6 +13,7 @@
 #include "httpd.h"
 #include "httpdespfs.h"
 #include "cgi.h"
+#include "cmd.h"
```
This shouldn't be needed here?

tve commented on esp-link/cgiwifi.c:
```diff
@@ -13,7 +13,9 @@ Cgi/template routines for the /wifi url.
 
+
 #include <esp8266.h>
+#include "cmd.h"
```
This shouldn't be included here...

tve reviewed (changes requested):
Looking nice overall! I made a slew of minor comments.

dannybackx committed:
Handle part of TVE's input already.

dannybackx committed:
Put functions in the right places to minimize superfluous includes.

dannybackx committed:
Fix a handlers.c error I missed because of missing include file dependencies.
Also revert a Makefile change (lwip library) accidentally committed.
Add a "make depend" target to avoid missing dependencies.

dannybackx committed:
More small edits after full review of my code.

dannybackx committed:
Allow to overrule the build tag from local.conf if you're not in a git tree.

tve commented on cmd/handlers.c:
```diff
@@ -197,3 +250,162 @@ cmdAddCallback(CmdPacket *cmd) {
+    uint8_t ix;
+    cmdPopArg(&req, &ix, 1);
+    len = cmdArgLen(&req);
+    pass = (char *)os_malloc(len+1);
```
where is this ever freed? it looks like a heap allocation is totally unnecessary...

dannybackx replied:
Just committed a fix

dannybackx replied:
Just committed a fix

tve replied:
oops, sorry, after writing the comments I decided to merge your commit and fix the issues myself. It's already all on master :-) (No guarantees that it builds or works, I had to go to bed...)

dannybackx replied:
No problem, I only spent a few minutes on it. Short enough not to notice that you'd already done it.
I'll check the build.

tve commented on cmd/handlers.c:
```diff
@@ -197,3 +250,162 @@ cmdAddCallback(CmdPacket *cmd) {
+    ssid[len] = 0;
+
+    len = cmdArgLen(&req);
+    pass = (char *)os_malloc(len+2);
```
where is this ever freed?

tve commented on cmd/handlers.c:
```diff
@@ -197,3 +250,162 @@ cmdAddCallback(CmdPacket *cmd) {
+    myssid[32] = '\0';
+    connectToNetwork(myssid, pass);
+  } else {
+    ssid = os_malloc(len+1);
```
where is this ever freed?

dannybackx replied:
Just committed a fix
